src/utils/dates.js

```diff
@@ -10,3 +12,101 @@ const ONE_DAY_IN_MILLISECONDS = 86400000;
 export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
+
+export const handleNextPurchaseDate = (
+	name,
+	currentDate,
+	dateCreated,
+	purchasesCount,
+	dateNextPurchased,
+	dateLastPurchased,
+) => {
+	console.log('-------------------------------------------');
+	console.log(`[${name}]`);
+
+	try {
+		// get purchase intervals and get new estimation for next purchase date
+		const purchaseIntervals = calculatePurchaseIntervals(
+			currentDate,
+			dateCreated,
+			dateNextPurchased,
+			dateLastPurchased,
+		);
+		const estimatedNextPurchaseDate = getNextPurchaseEstimate(
+			purchaseIntervals,
+			purchasesCount,
+		);
+
+		console.log(
+			`New next purchase date: ${estimatedNextPurchaseDate.toLocaleString()}`,
+		);
+		console.log('-------------------------------------------');
+
+		return estimatedNextPurchaseDate;
+	} catch (error) {
```

Comment:
Again as the comment about `FirestoreError` type, thrown errors are typically objects, so it's important to think about what types of errors you are anticipating encountering, and what property(ies) on the error object are most useful for your "wrapped" error you're passing along.

Reply:
These functions are invoked before reaching firebase, and I checked that <code>calculateEstimate</code> error doesn't change be it <code>error</code> or <code>error.message</code>!